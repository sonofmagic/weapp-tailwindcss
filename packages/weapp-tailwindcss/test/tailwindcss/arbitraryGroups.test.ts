import { getCss } from '../helpers/getTwCss'

describe('Arbitrary groups', () => {
  it('group case 0', async () => {
    const { css } =
      await getCss(`<a href="#" class="group block max-w-xs mx-auto rounded-lg p-6 bg-white ring-1 ring-slate-900/5 shadow-lg space-y-3 hover:bg-sky-500 hover:ring-sky-500">
    <div class="flex items-center space-x-3">
      <svg class="h-6 w-6 stroke-sky-500 group-hover:stroke-white" fill="none" viewBox="0 0 24 24"><!-- ... --></svg>
      <h3 class="text-slate-900 group-hover:text-white text-sm font-semibold">New project</h3>
    </div>
    <p class="text-slate-500 group-hover:text-white text-sm">Create a new project from a variety of starting templates.</p>
  </a>`)
    expect(css).toMatchSnapshot()
  })

  it('group case 1', async () => {
    const { css } = await getCss(`<ul role="list">
      {#each people as person}
        <li class="group/item hover:bg-slate-100 ...">
          <img src="{person.imageUrl}" alt="" />
          <div>
            <a href="{person.url}">{person.name}</a>
            <p>{person.title}</p>
          </div>
          <a class="group/edit invisible hover:bg-slate-200 group-hover/item:visible ..." href="tel:{person.phone}">
            <span class="group-hover/edit:text-gray-700 ...">Call</span>
            <svg class="group-hover/edit:translate-x-0.5 group-hover/edit:text-slate-500 ...">
              <!-- ... -->
            </svg>
          </a>
        </li>
      {/each}
    </ul>`)
    expect(css).toMatchSnapshot()
  })

  it('group case 2', async () => {
    const { css } = await getCss(`<div class="group is-published">
    <div class="hidden group-[.is-published]:block">
      Published
    </div>
  </div>`)
    expect(css).toMatchSnapshot()
  })

  it('group case 3', async () => {
    const { css } = await getCss(`<div class="group">
    <div class="group-[:nth-of-type(3)_&]:block">
      <!-- ... -->
    </div>
  </div>`)
    expect(css).toMatchSnapshot()
  })

  it('peer case 0', async () => {
    const { css } = await getCss(`<form>
    <label class="block">
      <span class="block text-sm font-medium text-slate-700">Email</span>
      <input type="email" class="peer ..."/>
      <p class="mt-2 invisible peer-invalid:visible text-pink-600 text-sm">
        Please provide a valid email address.
      </p>
    </label>
  </form>`)
    expect(css).toMatchSnapshot()
  })

  it('peer case 1', async () => {
    const { css } = await getCss(`<fieldset>
    <legend>Published status</legend>
  
    <input id="draft" class="peer/draft" type="radio" name="status" checked />
    <label for="draft" class="peer-checked/draft:text-sky-500">Draft</label>
  
    <input id="published" class="peer/published" type="radio" name="status" />
    <label for="published" class="peer-checked/published:text-sky-500">Published</label>
  
    <div class="hidden peer-checked/draft:block">Drafts are only visible to administrators.</div>
    <div class="hidden peer-checked/published:block">Your post will be publicly visible on your site.</div>
  </fieldset>`)
    expect(css).toMatchSnapshot()
  })

  it('peer case 2', async () => {
    const { css } = await getCss(`<form>
    <label for="email">Email:</label>
    <input id="email" name="email" type="email" class="is-dirty peer" required />
    <div class="peer-[.is-dirty]:peer-required:block hidden">This field is required.</div>
    <!-- ... -->
  </form>`)
    expect(css).toMatchSnapshot()
  })
})
