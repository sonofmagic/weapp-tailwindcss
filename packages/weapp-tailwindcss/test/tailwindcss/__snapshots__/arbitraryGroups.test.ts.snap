// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`arbitrary groups > group case 0 1`] = `
".mx-auto {
    margin-left: auto;
    margin-right: auto
}
.block {
    display: block
}
.flex {
    display: flex
}
.h-6 {
    height: 1.5rem
}
.w-6 {
    width: 1.5rem
}
.max-w-xs {
    max-width: 20rem
}
.items-center {
    align-items: center
}
.space-x-3 > :not([hidden]) ~ :not([hidden]) {
    --tw-space-x-reverse: 0;
    margin-right: calc(0.75rem * var(--tw-space-x-reverse));
    margin-left: calc(0.75rem * calc(1 - var(--tw-space-x-reverse)))
}
.space-y-3 > :not([hidden]) ~ :not([hidden]) {
    --tw-space-y-reverse: 0;
    margin-top: calc(0.75rem * calc(1 - var(--tw-space-y-reverse)));
    margin-bottom: calc(0.75rem * var(--tw-space-y-reverse))
}
.rounded-lg {
    border-radius: 0.5rem
}
.bg-white {
    --tw-bg-opacity: 1;
    background-color: rgb(255 255 255 / var(--tw-bg-opacity))
}
.stroke-sky-500 {
    stroke: #0ea5e9
}
.p-6 {
    padding: 1.5rem
}
.text-sm {
    font-size: 0.875rem;
    line-height: 1.25rem
}
.font-semibold {
    font-weight: 600
}
.text-slate-500 {
    --tw-text-opacity: 1;
    color: rgb(100 116 139 / var(--tw-text-opacity))
}
.text-slate-900 {
    --tw-text-opacity: 1;
    color: rgb(15 23 42 / var(--tw-text-opacity))
}
.shadow-lg {
    --tw-shadow: 0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1);
    --tw-shadow-colored: 0 10px 15px -3px var(--tw-shadow-color), 0 4px 6px -4px var(--tw-shadow-color);
    box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow)
}
.ring-1 {
    --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
    --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);
    box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow, 0 0 #0000)
}
.ring-slate-900\\/5 {
    --tw-ring-color: rgb(15 23 42 / 0.05)
}
.hover\\:bg-sky-500:hover {
    --tw-bg-opacity: 1;
    background-color: rgb(14 165 233 / var(--tw-bg-opacity))
}
.hover\\:ring-sky-500:hover {
    --tw-ring-opacity: 1;
    --tw-ring-color: rgb(14 165 233 / var(--tw-ring-opacity))
}
.group:hover .group-hover\\:stroke-white {
    stroke: #fff
}
.group:hover .group-hover\\:text-white {
    --tw-text-opacity: 1;
    color: rgb(255 255 255 / var(--tw-text-opacity))
}"
`;

exports[`arbitrary groups > group case 1 1`] = `
".invisible {
    visibility: hidden
}
.hover\\:bg-slate-100:hover {
    --tw-bg-opacity: 1;
    background-color: rgb(241 245 249 / var(--tw-bg-opacity))
}
.hover\\:bg-slate-200:hover {
    --tw-bg-opacity: 1;
    background-color: rgb(226 232 240 / var(--tw-bg-opacity))
}
.group\\/item:hover .group-hover\\/item\\:visible {
    visibility: visible
}
.group\\/edit:hover .group-hover\\/edit\\:translate-x-0 {
    --tw-translate-x: 0px;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))
}
.group\\/edit:hover .group-hover\\/edit\\:translate-x-0\\.5 {
    --tw-translate-x: 0.125rem;
    transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))
}
.group\\/edit:hover .group-hover\\/edit\\:text-gray-700 {
    --tw-text-opacity: 1;
    color: rgb(55 65 81 / var(--tw-text-opacity))
}
.group\\/edit:hover .group-hover\\/edit\\:text-slate-500 {
    --tw-text-opacity: 1;
    color: rgb(100 116 139 / var(--tw-text-opacity))
}"
`;

exports[`arbitrary groups > group case 2 1`] = `
".hidden {
    display: none
}
.group.is-published .group-\\[\\.is-published\\]\\:block {
    display: block
}"
`;

exports[`arbitrary groups > group case 3 1`] = `
":nth-of-type(3) .group .group-\\[\\:nth-of-type\\(3\\)_\\&\\]\\:block {
    display: block
}"
`;

exports[`arbitrary groups > peer case 0 1`] = `
".invisible {
    visibility: hidden
}
.mt-2 {
    margin-top: 0.5rem
}
.block {
    display: block
}
.text-sm {
    font-size: 0.875rem;
    line-height: 1.25rem
}
.font-medium {
    font-weight: 500
}
.text-pink-600 {
    --tw-text-opacity: 1;
    color: rgb(219 39 119 / var(--tw-text-opacity))
}
.text-slate-700 {
    --tw-text-opacity: 1;
    color: rgb(51 65 85 / var(--tw-text-opacity))
}
.peer:invalid ~ .peer-invalid\\:visible {
    visibility: visible
}"
`;

exports[`arbitrary groups > peer case 1 1`] = `
".visible {
    visibility: visible
}
.hidden {
    display: none
}
.peer\\/draft:checked ~ .peer-checked\\/draft\\:block {
    display: block
}
.peer\\/published:checked ~ .peer-checked\\/published\\:block {
    display: block
}
.peer\\/draft:checked ~ .peer-checked\\/draft\\:text-sky-500 {
    --tw-text-opacity: 1;
    color: rgb(14 165 233 / var(--tw-text-opacity))
}
.peer\\/published:checked ~ .peer-checked\\/published\\:text-sky-500 {
    --tw-text-opacity: 1;
    color: rgb(14 165 233 / var(--tw-text-opacity))
}"
`;

exports[`arbitrary groups > peer case 2 1`] = `
".hidden {
    display: none
}
.peer.is-dirty:required ~ .peer-\\[\\.is-dirty\\]\\:peer-required\\:block {
    display: block
}"
`;
